modules/reindex/src/test/java/org/elasticsearch/index/reindex/BulkByScrollTaskTests.java

```diff
@@ -20,281 +20,143 @@
+            int versionConflicts = between(0, 100);
+            int noops = total - updated - created - deleted;
+            int bulkRetries = between(0, 100);
+            int searchRetries = between(0, 100);
+            TimeValue throttled = timeValueNanos(between(0, 10000));
+            float requestsPerSecond = randomValueOtherThanMany(r -> r <= 0, () -> randomFloat());
+            String reasonCancelled = randomBoolean() ? null : "test";
+            TimeValue throttledUntil = timeValueNanos(between(0, 1000));
+            statuses[i] = new BulkByScrollTask.StatusOrException(new BulkByScrollTask.Status(i, total, updated, created, deleted, batches,
+                    versionConflicts, noops, bulkRetries, searchRetries, throttled, requestsPerSecond, reasonCancelled, throttledUntil));
+            mergedTotal += total;
+            mergedUpdated += updated;
+            mergedCreated += created;
+            mergedDeleted += deleted;
+            mergedBatches += batches;
+            mergedVersionConflicts += versionConflicts;
+            mergedNoops += noops;
+            mergedBulkRetries += bulkRetries;
+            mergedSearchRetries += searchRetries;
+            mergedThrottled = timeValueNanos(mergedThrottled.nanos() + throttled.nanos());
+            mergedRequestsPerSecond += requestsPerSecond;
+            mergedThrottledUntil = timeValueNanos(min(mergedThrottledUntil.nanos(), throttledUntil.nanos()));
+        }
+        String reasonCancelled = randomBoolean() ? randomAsciiOfLength(10) : null;
+        BulkByScrollTask.Status merged = new BulkByScrollTask.Status(Arrays.asList(statuses), reasonCancelled);
+        assertEquals(mergedTotal, merged.getTotal());
+        assertEquals(mergedUpdated, merged.getUpdated());
+        assertEquals(mergedCreated, merged.getCreated());
+        assertEquals(mergedDeleted, merged.getDeleted());
+        assertEquals(mergedBatches, merged.getBatches());
+        assertEquals(mergedVersionConflicts, merged.getVersionConflicts());
+        assertEquals(mergedNoops, merged.getNoops());
+        assertEquals(mergedBulkRetries, merged.getBulkRetries());
+        assertEquals(mergedSearchRetries, merged.getSearchRetries());
+        assertEquals(mergedThrottled, merged.getThrottled());
+        assertEquals(mergedRequestsPerSecond, merged.getRequestsPerSecond(), 0.0001f);
+        assertEquals(mergedThrottledUntil, merged.getThrottledUntil());
+        assertEquals(reasonCancelled, merged.getReasonCancelled());
+    }
 
-        int total = between(0, 1000000);
-        task.rethrottle(1);
-        assertThat((double) task.perfectlyThrottledBatchTime(total),
-                closeTo(TimeUnit.SECONDS.toNanos(total), TimeUnit.SECONDS.toNanos(1)));
+    public void testUnknownVersions() {
+        assertThat("5.1.0 has been defined, remove the temporary constant", VersionUtils.allVersions(),
+                not(hasItem(BulkByScrollTask.V_5_1_0_UNRELEASED)));
```

Comment:
I'd have to move the version constant away from where it is used to do that. I'm fine doing it, but the instructions we have suggest not doing that.